### PR TITLE
gmsh v4.8.4

### PR DESCRIPTION
**Pull request by matthiasdiener**
xref #40 

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by matthiasdiener** [comment]
Windows build error is:

```
[  3%] Building CXX object CMakeFiles/shared.dir/Common/gmsh.cpp.obj
gmsh.cpp
d:\bld\gmsh-packages_1619546255457\_h_env\library\include\fl\win32.H(128): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
d:\bld\gmsh-packages_1619546255457\_h_env\library\include\fl\win32.H(128): error C2146: syntax error: missing ';' before identifier 'fl_msg'
```

Line 128 in `include\fl\win32.H` is:

```c
extern FL_EXPORT MSG fl_msg;
```

Maybe `MSG` (https://docs.microsoft.com/en-us/windows/win32/api/winuser/ns-winuser-msg) gets overridden somewhere in FLTK or gmsh?

**Comment by isuruf** [comment]
See https://gitlab.onelab.info/gmsh/gmsh/-/merge_requests/421/diffs

**Comment by matthiasdiener** [comment]
This gets farther, the current win error is
```
[ 94%] Linking CXX executable gmsh.exe
LINK Pass 1: command "C:\PROGRA~2\MICROS~1\2017\ENTERP~1\VC\Tools\MSVC\1416~1.270\bin\Hostx64\x64\link.exe /nologo @CMakeFiles\gmsh.dir\objects1.rsp /out:gmsh.exe /implib:gmsh.lib /pdb:D:\bld\gmsh-packages_1619555788738\work\build\gmsh.pdb /version:0.0 /machine:x64 /debug /INCREMENTAL /subsystem:windows /STACK:16777216 /SAFESEH:NO gmsh.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\fltk_images.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\fltk_gl.lib opengl32.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\fltk.lib wsock32.lib comctl32.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\fltk_png.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\jpeg.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\z.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKXDESTEP.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKXDEIGES.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKXCAF.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKLCAF.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKVCAF.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKCAF.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKV3d.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKService.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKCDF.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\freetype.lib windowscodecs.lib ole32.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKSTEP.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKSTEP209.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKSTEPAttr.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKSTEPBase.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKIGES.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKXSBase.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKOffset.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKFeat.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKFillet.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKBool.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKMesh.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKHLR.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKBO.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKPrim.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKShHealing.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKTopAlgo.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKGeomAlgo.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKBRep.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKGeomBase.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKG3d.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKG2d.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKMath.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKernel.lib winmm.lib wsock32.lib ws2_32.lib psapi.lib opengl32.lib glu32.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\fltk.lib wsock32.lib comctl32.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\fltk_png.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\jpeg.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\z.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKXDESTEP.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKXDEIGES.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKXCAF.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKLCAF.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKVCAF.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKCAF.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKV3d.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKService.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKCDF.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\freetype.lib windowscodecs.lib ole32.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKSTEP.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKSTEP209.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKSTEPAttr.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKSTEPBase.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKIGES.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKXSBase.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKOffset.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKFeat.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKFillet.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKBool.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKMesh.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKHLR.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKBO.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKPrim.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKShHealing.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKTopAlgo.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKGeomAlgo.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKBRep.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKGeomBase.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKG3d.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKG2d.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKMath.lib D:\bld\gmsh-packages_1619555788738\_h_env\Library\lib\TKernel.lib winmm.lib ws2_32.lib psapi.lib glu32.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:CMakeFiles\gmsh.dir/intermediate.manifest CMakeFiles\gmsh.dir/manifest.res" failed (exit code 1120) with the following output:
LINK : gmsh.exe not found or not built by the last incremental link; performing full link
fltk.lib(fl_call_main.c.obj) : error LNK2019: unresolved external symbol main referenced in function WinMain
gmsh.exe : fatal error LNK1120: 1 unresolved externals
NMAKE : fatal error U1077: 'D:\bld\gmsh-packages_1619555788738\_build_env\Library\bin\cmake.exe' : return code '0xffffffff'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\bin\HostX64\x64\nmake.exe"' : return code '0x2'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\bin\HostX64\x64\nmake.exe"' : return code '0x2'
Stop.
```

File: recipe/patches/0001-wmain-main.patch [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+diff --git a/Common/Main.cpp b/Common/Main.cpp
+index 81eb10cc7..fc0d81d24 100644
+--- a/Common/Main.cpp
++++ b/Common/Main.cpp
+@@ -30,7 +30,7 @@ static char *toUTF8(wchar_t *src)
+   return out;
+ }
+ 
+-int wmain(int argc, wchar_t *wargv[], wchar_t *envp[])
++int main(int argc, wchar_t *wargv[], wchar_t *envp[])
```
Comment (isuruf) [inline recipe/patches/0001-wmain-main.patch:19]: This isn't correct though, `main` signature is `char*` and not `wchar_t*`.

Comment (isuruf) [reply recipe/patches/0001-wmain-main.patch:19]: I think we can avoid this by building fltk as a shared library where `winMain` is not defined.

Comment (matthiasdiener) [reply recipe/patches/0001-wmain-main.patch:19]: https://github.com/conda-forge/fltk-feedstock/pull/12

Comment (matthiasdiener) [reply recipe/patches/0001-wmain-main.patch]: I've implemented a simpler change (disabling wmain() when building with FLTK). Would that be ok?

**Comment by matthiasdiener** [comment]
This is ready for review @conda-forge/gmsh

**Comment by matthiasdiener** [comment]
@conda-forge-admin restart CI

**Comment by matthiasdiener** [comment]
@conda-forge-admin restart CI

Spurious Travis failures...

**Comment by matthiasdiener** [comment]
@conda-forge-admin restart CI

**Comment by matthiasdiener** [comment]
Pinging @conda-forge/gmsh  for review.

**Pull request merged**